Edit message index page

yuka-matsuura opened the pull request:
# What
チャット画面の実装

# Why
必須機能のビューとなるから

<img width="1440" alt="2018-03-01 18 22 39" src="https://user-images.githubusercontent.com/35907344/36837954-1bccc290-1d81-11e8-9d26-1e4ff6575d4a.png">

Commit:
Create chat view

# What
チャット画面の作成（view,controller,routing作成）

# Why
必須機能のため

Commit:
Change application stylesheets file format

# What
 Application.cssからscssのファイル形式に変更
また、読み込むスタイルの記述

# Why
Sassを使用してスタイルを設定するため

Commit:
Create chat view side-bar

# What
チャット画面のサイドバー（ユーザ情報とグループ情報一覧）作成

#Why
メイン機能の画面の一部だから

Commit:
comment out unnecessary messages

# What
コメントアウトの方法が間違っていたので修正

Commit:
Change chat view

# What
 チャット画面の実装

# Why
必須機能のビューとなるから

yamamotokeishiro commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,21 @@
+    .messages
+      =render partial: "messages/message"
+    %form.message-form
+      %input.message-text{ :placeholder => "type a message" }
```
ハッシュロケットは現在推奨されなくなって来ておりますので、ハッシュロケットを使わない記述にしていきましょう！

yamamotokeishiro commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,21 @@
+      .image-form
+        .image-form__message-image
+          %i.fa.fa-picture-o.icon
+      %input.form-submit{ :type => "submit", :value => "Send" }
```
ハッシュロケットは現在推奨されなくなって来ておりますので、ハッシュロケットを使わない記述にしていきましょう！

yamamotokeishiro commented on Gemfile:
```diff
@@ -54,8 +54,7 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'font-awesome-rails'
 
 
 
```
下の空白も削除しましょう！

yamamotokeishiro commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,155 @@
+          height: 100px;
+          padding: 0 20px;
+          color: #FFFFFF;
+          .user-information__user-name{
```
BEM記法で、＆を使用することによりコードの省略化が可能ですので省略出来る部分は使用しましょう！
例）.user-information{
　　　＆__user-name{

http://taneppa.net/css_bem/
こちらの記事がオススメです！

yamamotokeishiro commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,155 @@
+        .groups{
+          background-color: #2f3e51;
+          padding: 0 20px;
+          color: #FFFFFF;
```
同じカラーを複数使用する場合は、変数化してまとめることが出来ます！
複数出てくるカラーは変数化しましょう！

yamamotokeishiro commented:
インデントが綺麗に揃っております！
その他、微修正お願いいたします！

Commit:
Create message model

# What
Messageモデルを先走って作ってしまいました。
コミット忘れてたので、コミットしています。

# Why
本アプリケーションの必須機能のため

Commit:
Change Description without hash rocket

# What
ハッシュロケットの記述をシンボル型に書き換えました

# Why
ハッシュロケットは現在推奨されていない記述のため

Commit:
Delete unnecessary empty lines

# What
Gemfileに不要な空行があったので削除

Commit:
Change class name

# What
一部クラス名の修正

# Why
BEM記法の「＆」を使用することによりコードを省略化

Commit:
create variable color code

# What
カラーコードの変数化

Commit:
Change message view

# What
メッセージ投稿日時のpaddingを修正

yuka-matsuura commented:
ご指摘いただいた事項全てに対応いたしました。
お手数をおかけしますが、ご確認の程よろしくお願いいたします。

takeru56 commented on app/views/messages/_group.html.haml:
```diff
@@ -0,0 +1,6 @@
+.group
+  .group__name group1
```
こちらも&をつかって省略できそうです.

例）.user-information{
　　　＆__user-name{

takeru56 commented on app/views/messages/_message.html.haml:
```diff
@@ -0,0 +1,12 @@
+.message
+  .upper-message
+    .upper-message__user-name user-name
```
同じくこちらも&をつかって省略できそうです.

例）.user-information{
　　　＆__user-name{

taketnk commented on app/assets/stylesheets/_messages.scss:
```diff
@@ -0,0 +1,162 @@
+$white: #FFFFFF;
```
こちら1~5行目
のカラー変数は、別ファイルに切り出して、importして呼び出すようにしておきましょう.
後々、scssの規模が大きくなった際に可読性が上がります.

Commit:
Cutting out color variables

# What
Messages.scssで使用していたカラー変数をcolors.scssに切り出して、呼び出す。

# Why
可読性の向上

yuka-matsuura commented:
レビュアーから、カラー変数を別ファイルに切り出す対応のみでよいと確認が取れているため、その対応のみ完了いたしました。
お手数をおかけしますが、ご確認の程よろしくお願いいたします。

taketnk commented on db/migrate/20180227011005_create_messages.rb:
```diff
@@ -0,0 +1,11 @@
+      t.string  :image
+      t.text    :content
+      t.integer :user, null:false, foreign_key:true
+      t.integer :group, null:false,foreign_key:true
```
`t.integer :user, null:false, foreign_key:true`
` t.integer :group, null:false,foreign_key:true`

これだとメッセージテーブルにuserカラムとgroupカラムが作成されてしまう気がするのですが、
必要なのは、user_idカラムとgroup_idカラムではないでしょうか？
確認してみてください。

yuka-matsuura replied:
ご確認ありがとうございます。
外部キーの設定が誤っておりました。修正いたします。

Commit:
Remove Unnecessary migration and model file

# What
Message投稿画面のビュー作成では、
不要なマイグレーションファイルとモデルファイルの削除

# Why
現段階では必要のないファイル。
※ご指摘いただいた内容は、usersテーブルやgroupsテーブル作成後に作る中間テーブルにて対応いたします。

yuka-matsuura commented:
ご指摘いただいた内容は後ほど対応いたしますので、現段階では不要なモデルとマイグレーションファイルの削除を行いました。
